AIPscan/helpers.py

```diff
@@ -10,6 +11,21 @@ def parse_bool(val, default=True):
         return default
 
 
+def parse_date(date_string):
+    """Parse date parameter string into datetime object or None.
+
+    :param date_string: Date string. Valid format is YYYY-MM-DD.
+
+    :returns: datetime.datetime object or None
+    """
+    if isinstance(date_string, str):
+        try:
+            return datetime.strptime(date_string, "%Y-%m-%d")
+        except ValueError:
+            pass
+    return None
```

Comment:
I suspect you can get away with something like:
```python
def parse_date(date_string):
    """Parse date parameter string into datetime object or None."""
    try:
        return datetime.strptime(date_string, "%Y-%m-%d")
    except (ValueError, TypeError):
        pass
```
If you wanted to simplify this a little if the function is kept as is.